src/components/themed/FilledTextInput.tsx

```diff
@@ -556,15 +576,24 @@ const StyledNumericInput = styledWithRef(NumericInput)<{
   disableAnimation: SharedValue<number>
   focusAnimation: SharedValue<number>
   scale: SharedValue<number>
+  dynamicHeight: SharedValue<number | undefined>
   textsizeRem?: number
-}>(theme => ({ disableAnimation, focusAnimation, textsizeRem, scale }) => {
+}>(theme => ({ disableAnimation, dynamicHeight, focusAnimation, textsizeRem, scale }) => {
   const rem = theme.rem(textsizeRem ?? 1)
   const interpolateTextColor = useAnimatedColorInterpolateFn(theme.textInputTextColor, theme.textInputTextColorFocused, theme.textInputTextColorDisabled)
   // Need 2 pixels of shift given a 16 point rem settings
   // This is due to Android rendering a text input vertically lower
   // than a Text field by ~2 pixels
   const androidVShift = isAndroid ? rem / 8 : 0
 
+  // For animating height changes in multiline input
+  const animatedHeightStyle = useAnimatedStyle(() => {
+    if (dynamicHeight.value == null) return {}
+    return {
+      height: withTiming(dynamicHeight.value, { duration: 100 }) // Smooth animation for height change
```

Comment:
This works great, and feels really smooth! However, if you type in more text than fits on the screen, the text input just keeps growing forever, and never gets a scroll bar. This makes it impossible to edit the top lines of text, because they get pushed off screen.

So, I think we either need a `flexShrink` or a `maxHeight` or something like that, so the text input does not overflow the screen.

Reply:
Ok looks like I way overcomplicated it. Trying to cap the height ends up in a circular dependency nightmare, and trying to guess how tall an input can be beforehand isn't really feasible across devices.

Just gonna table the original implementation, might come in handy later. For now, just making it always full screen for multiline.